src/crypto/tls/cfkem.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2022 Cloudflare, Inc. All rights reserved. Use of this source code
+// is governed by a BSD-style license that can be found in the LICENSE file.
+//
+// Glue to add Circl's (post-quantum) hybrid KEMs to quic-go.
+//
+// To enable set CurvePreferences with the desired scheme as the first element:
+//
+//   import (
+//      "github.com/cloudflare/circl/kem/tls"
+//      "github.com/cloudflare/circl/kem/hybrid"
+//
+//          [...]
+//
+//   config.CurvePreferences = []tls.CurveID{
+//      hybrid.Kyber512X25519().(tls.TLSScheme).TLSCurveID(),
+//      qtls.X25519,
+//      qtls.P256,
+//   }
+
+package tls
+
+import (
+	"fmt"
+	"io"
+
+	"circl/kem"
+	"circl/kem/hybrid"
+)
+
+// Either ecdheParameters or kem.PrivateKey
+type clientKeySharePrivate interface{}
+
+var (
+	kyber512X25519CurveID = CurveID(0xFF01)
+	kyber768X25519CurveID = CurveID(0xFF02)
+	invalidCurveID        = CurveID(0xFFFF)
```

Comment:
ID 65535 (0xFFFF) is not reserved, should we use the special value 0 instead or would that have other implications?
```suggestion
	invalidCurveID        = CurveID(0x0000)
```
"The value 0 (0x0000) has been marked as reserved." from https://www.rfc-editor.org/rfc/rfc8447.html#section-9